app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,26 @@
+class Admin::VideosController < ApplicationController
+  before_action :require_login
+  before_action :require_admin
+
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.create(video_params)
+
+    if @video.save
+      flash[:success] = "Your video #{@video.title} has been added"
+      redirect_to new_admin_video_path
+    else
+      flash[:danger] = "Please retry your entry"
```

Comment:
Remember its `flash.now` when you render. Otherwise, flash message pops on next GET request as well and you don't want to have error message there.